### DISCUSSION TITLE
Aesgis256 for Encryption

### DISCUSSION THREAD

**Discussion by sidielnikov** [samuel-lucas6/Kryptor#57] [General]
Hello in the roadmap, you wrote that "Consider Aesgis256 for Encryption" I want to still still want to replace Chacha20 with Aesgis256 or how do you want to use it in Kryptor? I want to add that I'm really looking forward to support by Yubikeys in Kryptor

**Comment by samuel-lucas6** [comment]
Good question. Yes, it would be to replace ChaCha20-Poly1305 because it's far more performant and key committing. However, it hasn't become an RFC yet, isn't fully committing, and libsodium 1.0.19 will likely be a while. It may never happen.

As for YubiKey support, I would like to reiterate that that's a long way off. It doesn't sound like something that can be done in an afternoon.

My primary focus has to be my MSc course, my dissertation will be ongoing throughout the summer when I've previously had holiday, I'd like to continue implementing more algorithms for learning purposes, there are other tools/libraries to maintain and ideas to code up, and so on. Everything is a battle for time.

**Reply by sidielnikov** [reply]
In what issue do you plan to add Aesgis256?

**Reply by samuel-lucas6** [reply]
I'm afraid I don't know what you mean. It would be a breaking change and replace ChaCha20-Poly1305 everywhere it's currently being used.
